clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -934,6 +935,44 @@ mlir::SuccessorOperands BrOp::getSuccessorOperands(unsigned index) {
 
 Block *BrOp::getSuccessorForOperands(ArrayRef<Attribute>) { return getDest(); }
 
+static void eraseBlock(Block *block) {
+  for (auto &op : llvm::make_early_inc_range(*block))
+    op.erase();
+  block->erase();
+  return;
+}
+
+/// Removes branches between two blocks if it is the only branch.
+///
+/// From:
+///   ^bb0:
+///     cir.br ^bb1
+///   ^bb1:  // pred: ^bb0
+///     cir.return
+///
+/// To:
+///   ^bb0:
+///     cir.return
+LogicalResult BrOp::fold(FoldAdaptor adaptor,
+                         SmallVectorImpl<OpFoldResult> &results) {
+  Block *block = getOperation()->getBlock();
+  Block *dest = getDest();
+
+  if (getOperation()->getParentOfType<mlir::LLVM::LLVMFuncOp>() ||
```

Comment:
I don't understand why you are looking at `mlir::LLVM::LLVMFuncOp` here, there should be no LLVM dialect reference in CIRDialect. We only care about folding CIR stuff, LLVM folding should be handled by LLVM stuff and should be unrelated to CIR.